internal/platform/writer/console.go

```diff
@@ -21,13 +21,35 @@ var _ scan.Writer = Console{}
 // Console is a [scan.Writer] implementation that writes the output
 // to the given [io.Writer], following console/terminal standards
 // in a human-friendly fashion.
+
+type ConsoleOption func(*Console)
+
 type Console struct {
-	writer io.Writer
+	writer     io.Writer
+	pocEnabled bool
+}
+
+// WithPOCEnabled is a [ConsoleOption] that enables the Proof of Concept (PoC) mode.
+// When enabled, the console will print the matches in a more readable format.
+// This is useful when the user wants to copy the PoC to a file or another tool.
+func WithPOCEnabled(enabled bool) func(*Console) {
+	return func(c *Console) {
+		c.pocEnabled = enabled
+	}
 }
 
 // NewConsole creates a new instance of [Console] with the given [io.Writer].
-func NewConsole(writer io.Writer) Console {
-	return Console{writer: writer}
+func NewConsole(writer io.Writer, opts ...ConsoleOption) Console {
```

Comment:
Much better now, that polluting the writer interface 👏🏻 Thanks! 🙌🏻

Reply:
you're most welcome